h2/src/main/org/h2/command/Parser.java

```diff
@@ -4866,9 +4866,11 @@ private AlterIndexRename parseAlterIndex() {
 
     private AlterView parseAlterView() {
         AlterView command = new AlterView(session);
+        boolean ifExists = readIfExists(false);
+        command.setIfExists(ifExists);
         String viewName = readIdentifierWithSchema();
         Table tableView = getSchema().findTableOrView(session, viewName);
-        if (!(tableView instanceof TableView)) {
+        if (!(tableView instanceof TableView) && !ifExists) {
             throw DbException.get(ErrorCode.VIEW_NOT_FOUND_1, viewName);
         }
         TableView view = (TableView) tableView;
```

Comment:
I had thought about this but I think that returning `NoOperation` early is wrong because syntax errors that arise later on would be swallowed early. E.g. currently, the `RECOMPILE` keyword is mandatory, so we must parse it.

An option would be to rearrange the statements and finish parsing first, before checking semantic errors.